src/coins.cpp

```diff
@@ -110,10 +110,16 @@ void CCoinsViewCache::AddCoin(const COutPoint &outpoint, Coin&& coin, bool possi
            (bool)it->second.coin.IsCoinBase());
 }
 
-void CCoinsViewCache::EmplaceCoinInternalDANGER(COutPoint&& outpoint, Coin&& coin) {
-    cachedCoinsUsage += coin.DynamicMemoryUsage();
-    auto [it, inserted] = cacheCoins.try_emplace(std::move(outpoint), std::move(coin));
-    if (inserted) CCoinsCacheEntry::SetDirty(*it, m_sentinel);
+void CCoinsViewCache::EmplaceCoinInternalDANGER(COutPoint&& outpoint, Coin&& coin)
+{
+    auto [it, inserted] = cacheCoins.try_emplace(std::move(outpoint));
```

Comment:
I think this would be simpler if we kept the original behavior of moving the `coin` here as well, and then only increment the `cachedCoinsUsage` if the coin is inserted.

In practice it's not possible to get `!inserted` unless the assume utxo payload was generated incorrectly.

Reply:
> In practice it's not possible to get !inserted unless the assume utxo payload was generated incorrectly.

As far as I can see that's not a guarantee, but I've also noted it in the commit message that currently this is the case.

> kept the original behavior of moving the coin here as well

Sure, if you think that's better - changed!